pyproject.toml

```diff
@@ -82,7 +83,8 @@ before-all = [
   "git clone --depth 1 --branch v1.9.4 https://github.com/lz4/lz4 lz4",
   "cd lz4",
   "make",
-  "make install DESTDIR=/tmp/lz4_install"
+  "make install PREFIX=$LZ4_INSTALL_DIR",
+  "find $LZ4_INSTALL_DIR",
```

Comment:
Debugging step?

Reply:
Yeah, but let's leave it in. It's not noisy, and having it in the logs seems better than not, the next time we've got to debug this.

Reply:
I don't get why delocate is requiring us to set `DYLD_LIBRARY_PATH` at all, though. It should be following the rpath we're setting to find the right library, but it's not. Either we're not setting the rpath right (though all the linker command lines look right to me) or delocate is for some reason ignoring that rpath in favor of the system libs...

Reply:
I don't understand why this is biting us right now

Reply:
It's broken on main, too - the only difference is that the new cibuildwheel fails when delocate produces a wheel that doesn't conform to the declared `MACOSX_DEPLOYMENT_TARGET`, while the old one succeeds despite the produced wheel being incompatible with our declared target. Looks like it's always been wrong (or at least, has been wrong for a while), but we're being "bitten" now because the new version notices that something is wrong.

Reply:
[The log](https://github.com/bloomberg/memray/actions/runs/9062564159/job/24896720712#step:8:1807) on `main` shows:
```text
  + delocate-wheel --require-archs x86_64 -w /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-py7ipllh/cp38-macosx_x86_64/repaired_wheel -v /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-py7ipllh/cp38-macosx_x86_64/built_wheel/memray-1.12.0-cp38-cp38-macosx_10_14_x86_64.whl
  INFO:delocate.delocating:Copying library /usr/local/Cellar/lz4/1.9.4/lib/liblz4.1.9.4.dylib to memray/.dylibs/liblz4.1.9.4.dylib
  INFO:delocate.delocating:Modifying install name in memray/_memray.cpython-38-darwin.so from /usr/local/lib/liblz4.1.dylib to @loader_path/.dylibs/liblz4.1.9.4.dylib
  Fixing: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-py7ipllh/cp38-macosx_x86_64/built_wheel/memray-1.12.0-cp38-cp38-macosx_10_14_x86_64.whl
  Copied to package .dylibs directory:
    /usr/local/Cellar/lz4/1.9.4/lib/liblz4.1.9.4.dylib
```

On this branch, [that's now fixed](https://github.com/bloomberg/memray/actions/runs/9057804478/job/24882363005?pr=595#step:8:1843) to be:
```text
  + DYLD_LIBRARY_PATH=/tmp/lz4_install/lib delocate-wheel --require-archs x86_64 -w /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-o8x9exl4/cp38-macosx_x86_64/repaired_wheel -v /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-o8x9exl4/cp38-macosx_x86_64/built_wheel/memray-1.12.0-cp38-cp38-macosx_10_14_x86_64.whl
  INFO:delocate.delocating:Copying library /private/tmp/lz4_install/lib/liblz4.1.9.4.dylib to memray/.dylibs/liblz4.1.9.4.dylib
  INFO:delocate.delocating:Modifying install name in memray/_memray.cpython-38-darwin.so from /usr/local/lib/liblz4.1.dylib to @loader_path/.dylibs/liblz4.1.9.4.dylib
  Fixing: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-o8x9exl4/cp38-macosx_x86_64/built_wheel/memray-1.12.0-cp38-cp38-macosx_10_14_x86_64.whl
  Copied to package .dylibs directory:
    /private/tmp/lz4_install/lib/liblz4.1.9.4.dylib
```